### DISCUSSION TITLE
SerialLog 0 does not appear to disable the hardware serial bridge as stated in documentation

### DISCUSSION THREAD

**Discussion by MTGS69** [arendst/Tasmota#18049] [Q&A] [answered]
The documentation states : "_Note that changing serial logging (SerialLog 0) will disable the hardware Serial Bridge._"  However, this does not appear to be the case.  I am running Tasmota 12.4.0 on a Wemos D1 Mini.  I have one of those relatively new human sensors attached via hardware serial to GPIO 1 and 3  (module set with GPIO1 = Serial Tx and GPIO3 = Serial RX).  I use the serial interface to set/view the sensor settings from within the web terminal interface.  However, the device outputs a large amount of "SerialReceived" data when it senses motion (see picture below).  Therefore, I'd like to disable the serial bridge once tuning is complete as the sensor also has a simple high/low output pin to indicate motion.  That is, I only need the serial bridge for occasional tuning).  I have all logging disabled (except WebLog = 2) but have been unable to suppress these messages.  I can do it with the software serial bridge but it is not able to keep up with output resulting in occasional garbled output and buffer overruns.  Is there any way to suppress hardware SerialReceived messages?  My option of last resort is to add a transistor switch to physically activate/deactivate the actual TX connection.  Any advice is most appreciated.....thanks in advance.
![serial snip](https://user-images.githubusercontent.com/32232209/221350870-ad052627-2f45-4e83-a292-6cc9c897f799.png)

**Comment by sfromis** [comment]
One way of not having Tasmota treat serial input as serial would be to config RX as something else. Maybe `Output Lo` where Tasmota will just have the pin low.

**Reply by barbudor** [reply]
Reconfiguring an input to an output without disconnecting the incoming signal is not a good idea.
None would be safer

**Reply by sfromis** [reply]
"None" does not work for that, as Tasmota automatically assumes serial.

**Reply by barbudor** [reply]
So anything that is an input

@MTGS69 where did you read that in the doc? 
Obviously SerialLog 0 is required to use SerialBride not the other way round (using SerialSend5 for example will automatically set SerialLog to 0)

**Reply by sfromis** [reply]
The quote is from here:
https://tasmota.github.io/docs/Commands/#serial-bridge

Normally, Tasmota will not start listening for `SerialReceived` until a `SerialSend` has been done after each boot, as detailed in the next paragraph.

**Reply by MTGS69** [reply]
@barbudor, the link sfromis provided is indeed where I found that statement in the docs.  I will say, I was surprised a logging command was used to enable/disable the serial bridge but that is all I have found on the subject.  BTW.....and in case it is not obvious, I do not care so much about logging messages.  Rather, I'm more concerned with flooding my MQTT server with unnecessary messages. @sfromis I have the following rule to configure the serial settings on boot.  Although not ideal, if no messages are sent unless configured, that could be an option (i.e. disable this rule).  However, rather than having to restart and manually setup the bridge settings, I'm probably more inclined to go with the transistor switch option (have 6 installed and another 10 in the works).  `ON System#Boot Do Backlog Baudrate 115200; SerialConfig 3; SerialDelimiter 128; SerialBuffer 520; SerialSend get_all ENDON`

**Reply by sfromis** [reply]
You do not need this every boot:
`Backlog Baudrate 115200; SerialConfig 3; SerialDelimiter 128; SerialBuffer 520`

Of course, if you need that `SerialSend get_all`, this is where you enable Serial Bridge.

You could likely also use software serial (just using other than the hardware serial pins), where Tasmota does not default any serial config if pins (like RX) are just not configured.

**Reply by barbudor** [reply]
The text is obviously wrong
Using SerialBridge with a SerialSend would force SerialLog 0
The same, SerialLog 0 enable SerialReceive

**Reply by MTGS69** [reply]
@sfromis you are correct sir.  The rule simply requires the SerialSend command....leftover cruft from when I was first figuring this out.  Also, I tried to use the software serial bridge but it is too slow to keep up with the sensor.  @barbudor I figured that was the case (incorrect documentation) but thought I would ask before implementing a hardware solution.  Adding a transistor and resistor into the equation is not a big deal and works just fine.  I appreciate both of your responses.  Cheers!

**Reply by MTGS69** [reply]
I'm going to mark this as "answered".

**Comment by MTGS69** [comment]
BTW....I recommended to the vendor (Hi-Link) they implement a setting to limit the output of the serial interface.  The seemed to agree with my argument for doing so and I have seen this option in other of their products, but who knows if they will actually do so.

**Reply by Jason2866** [reply]
The sensor has a Tasmota driver ;-)

**Reply by MTGS69** [reply]
@Jason2866 I have the HLK-LD1115H sensors (not the 2410).  Based on the manual posted on Hi-Link website, the serial commands, baud rate, and tuning software (which I do not use) are different.  I cannot tell from the photos available, but based on the cost delta I suspect the 2410 uses a different microcontroller (not a STM).  I could be wrong and the Tasmota driver may still be compatible.  I shall investigate. Thanks for the tip!

**Comment by MTGS69** [comment, marked as answer]
I believe this is the most accurate answer to the specific question asked:

https://github.com/arendst/Tasmota/discussions/18049#discussioncomment-5108267
